### ISSUE TITLE
Support video and others audio formats

### ISSUE THREAD

**Issue by Daniel20Ultrastar-es** [SarutaSan72/Yass#11]
This is a hard one. Currently yass doesn't have video support. As ultrastar can reproduce the audio stream from the video. Yass should do the same.

It will be a huge progress to creators, if they could create a song without a mp3 track, just with the video. Specially with the new formats mp4 & webm

Would be nice also the support of other audio streams such as ogg, flac, acc...

**Comment by SarutaSan72** [comment]
I started implementing this, but gave up. Not easy to do this properly with Java, at least back in 2010. If someone wants to follow up: you can still find video playback in the code, I have just deactived it.

IMO it is easier to include a third-party tool to extract audio and convert it to mp3, on-the-fly. Same for ogg, flac, acc.

**Comment by Daniel20Ultrastar-es** [comment]
extracting audio and convert to mp3 is not a nice solution because mp3 **ALWAYS** loss the quality.

Syncronize a song with the maximum quality is essential for creators. Some songs are hard to follow the rythm.

Also, you have to delete the audio from video, because if you don't do it, you'll need more space (mp3 file + audio track from video).

Let's see if we can do something with this.

----------------------------------------------------------
I was going to open another issue, but I'll ask you here, do you think is it easy to implement a new feature to slow down the speed of the mp3? (for example in youtube, you can play a video at 0.75 / 0.50 speed.) This would be an awesome feature too for creators.

**Comment by SarutaSan72** [comment]
True :) 

As for the speed-slowdown, see the recording function (press Ctrl+R).
There I use `YassPlayer.playSelection(in, out, clicks, timebase)`
with timebase = 1,2,3,4 for speed = 100%, 50%, 33%, 25%.
Seems doable. I've opened an issue for this.

**Comment by SarutaSan72** [comment]
I labeled this issue with "help wanted", since it is very unlikely that I will spend more time with this.

**Comment by TeLiXj** [comment]
I have an idea about this: if you open a song that are using the video's audio you can use ffmpeg library to create a temp mp3 file and internally use it. I found some java wrappers to use it. Load a video -> export to mp3: easy :D
How do you think about this?

**Comment by SarutaSan72** [comment]
Ten years ago, I had many problems with integrating ffmpeg and had to drop it. I tried to show video while editing. If we limit ourselves to extracting a temp mp3 file for internal use, and if today there is a stable java wrapper that supports the various video/audio formats, this might be doable. 

Personally, I'd prefer to dodge that challenge. For now, I'll switch back from wontfix to helpwanted.

**Comment by TeLiXj** [comment]
I'm going to try to do it... I'll need some luck ;D

**Comment by Tuupertunut** [comment]
Ogg Vorbis support for Java seems to be quite complete.

https://github.com/Trilarion/java-vorbis-support

**Comment by cyberbeat** [comment]
ffmpeg support: https://github.com/bytedeco/javacv?tab=readme-ov-file
